torch/utils/checkpoint.py

```diff
@@ -17,7 +21,7 @@ def detach_variable(inputs):
 
 
 def check_backward_validity(inputs):
-    if not any(inp.requires_grad for inp in inputs):
+    if not any(inp.requires_grad for inp in inputs if isinstance(inp, torch.Tensor)):
```

Comment:
I'm receiving the warning when I pass in a list of tensors which have gradients. Propose:
```
if not any(inp.requires_grad for inp in inputs if inputs is not None):
```

Reply:
@oliland send us a PR?